lib/hanami/cli/commands/destroy/app.rb

```diff
@@ -0,0 +1,119 @@
+
+            recursively_destroy_precompiled_assets(context)
+            destroy_assets_manifest(context)
+
+            recursively_destroy_specs(context)
+            recursively_destroy_app(context)
+          end
+
+          private
+
+          def assert_valid_app!(context)
+            return if project.app?(context)
+
+            existing_apps = project.apps.map { |name| "`#{name}'" }.join(' ')
+            warn "`#{context.app}' is not a valid APP. Please specify one of: #{existing_apps}"
+            exit(1)
+          end
+
+          def remove_test_http_session_secret(context)
+            content     = "#{context.app.upcase}_SESSIONS_SECRET"
+            destination = project.env(context, "test")
+
+            files.remove_line(destination, content)
+            say(:subtract, destination)
+          end
+
+          def remove_development_http_session_secret(context)
+            content     = "#{context.app.upcase}_SESSIONS_SECRET"
+            destination = project.env(context, "development")
+
+            files.remove_line(destination, content)
+            say(:subtract, destination)
+          end
+
+          def remove_mount_app(context)
+            content     = "mount #{context.app.classify}::Application"
+            destination = project.environment(context)
+
+            files.remove_line(destination, content)
+            say(:subtract, destination)
+          end
+
+          def remove_require_app(context)
+            content     = "require_relative '../apps/#{context.app}/application'"
+            destination = project.environment(context)
+
+            files.remove_line(destination, content)
+            say(:subtract, destination)
+          end
+
+          def recursively_destroy_precompiled_assets(context)
+            destination = project.public_app_assets(context)
+            return unless files.directory?(destination)
+
+            files.delete_directory(destination)
+            say(:remove, destination)
+          end
+
+          def destroy_assets_manifest(context)
+            destination = project.assets_manifest(context)
+            return unless files.exist?(destination)
+
+            files.delete(destination)
+            say(:remove, destination)
+          end
+
+          def recursively_destroy_specs(context)
+            destination = project.app_spec(context)
+
+            files.delete_directory(destination)
+            say(:remove, destination)
+          end
+
+          def recursively_destroy_app(context)
+            destination = project.app(context)
+
+            files.delete_directory(destination)
+            say(:remove, destination)
+          end
+
+          def base_url(context)
+            content     = "mount #{context.app.classify}::Application"
+            destination = project.environment(context)
+
+            line  = files.read_matching_line(destination, content)
+            *, at = line.split(/at\:[[:space:]]*/)
```

Comment:
@kaikuchn thank you!! IMO if we can replace that with your suggestion, it would improve the read of the code.